File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i = 0
        j = len(nums) - 1
        while i <= j:
            mid = i + ((j - i) // 2)
            print(nums[i])
            print(nums[j])
            print(nums[mid])
            if nums[mid] == target:
                return mid
            if nums[i] == target:
                return i
            if nums[j] == target:
                return j
            elif nums[i] < nums[mid] and target > nums[i] and target < nums[mid]:
                print("#")
                j = mid - 1
            elif nums[mid] < nums[j] and target > nums[mid] and target < nums[j]:
                print("##")
                i = mid + 1
            elif nums[i] > nums[mid]:
                print("###")
                j = mid - 1
            else:
                print("####")
                i = mid + 1
        return -1

        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, ArrayList<Integer>> frequencyMapList = new HashMap<>();
        HashMap<Integer, Integer> frequencyMapInteger = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++) {
            if(frequencyMapInteger.containsKey(nums[i])) {
                frequencyMapInteger.put(nums[i], frequencyMapInteger.get(nums[i]) + 1);
            }
            else {
                frequencyMapInteger.put(nums[i], 1);
            }
        }
        for(Integer element:frequencyMapInteger.keySet()) {
            if(frequencyMapList.containsKey(frequencyMapInteger.get(element))) {
                frequencyMapList.get(frequencyMapInteger.get(element)).add(element);
            }
            else {
                frequencyMapList.put(frequencyMapInteger.get(element),new ArrayList<Integer>());
                frequencyMapList.get(frequencyMapInteger.get(element)).add(element);
            }
        }
        int[] mainList = new int[frequencyMapInteger.size()];
        int index = 0;
        for(Integer element:frequencyMapList.keySet()) {
            for(Integer elementOfList:frequencyMapList.get(element)) {
                mainList[index] = elementOfList;
                index++;
            }
        }
        mainList = Arrays.copyOfRange(mainList, mainList.length-k, mainList.length);
        return mainList;
    }
}
